### PR TITLE
Add material ui support for rendering select lists in cells based on …

### PR DESCRIPTION
**Pull request by ryan-bennett**
Add material ui support for rendering select lists in cells based on oneOf schema definitions.

### CODE REVIEW

**Comment by sdirix** [comment]
Hi @ryan-bennett, thank you very much for the contribution :heart:  We'll have a look at it soon!

However, please remove the changes to the `package-lock.json` from your PR.

**Comment by ryan-bennett** [comment]
Thanks Stefan, removed the changes to the package-lock.

File: packages/react/src/JsonFormsContext.tsx [hunk 1]
```diff
@@ -403,6 +411,15 @@ const withContextToEnumProps =
       return (<Component {...props} {...dispatchProps} {...stateProps} options={stateProps.options} />);
     };
 
+const withContextToOneOfEnumCellProps =
+  (Component: ComponentType<EnumCellProps>): ComponentType<OwnPropsOfEnumCell> =>
+    // @ts-ignore
```
Comment (sdirix) [inline packages/react/src/JsonFormsContext.tsx:416]: Is this needed?
```suggestion
```

File: packages/react/src/JsonFormsContext.tsx [hunk 2]
```diff
@@ -403,6 +411,15 @@ const withContextToEnumProps =
...
+    ({ctx, props}: JsonFormsStateContext & EnumCellProps) => {
```
Comment (sdirix) [inline packages/react/src/JsonFormsContext.tsx:417]:
```suggestion
    ({ ctx, props }: JsonFormsStateContext & EnumCellProps) => {
```

File: packages/material/src/cells/MaterialOneOfEnumCell.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+} from '@jsonforms/core';
+import { withJsonFormsOneOfEnumCellProps } from '@jsonforms/react';
+import { MuiSelect } from '../mui-controls/MuiSelect';
+
+export const MaterialOneOfEnumCell = (props: EnumCellProps & WithClassname) => (
+  <MuiSelect {...props} />
+);
+
+/**
+ * Default tester for enum controls.
```
Comment (sdirix) [inline packages/material/src/cells/MaterialOneOfEnumCell.tsx:41]:
```suggestion
 * Default tester for oneOf enum controls.
```

File: packages/material/test/renderers/MaterialOneOfEnumCell.test.tsx [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+              {
+                const: 'AU',
+                title: 'Australia'
+              },
+              {
+                const: 'NZ',
+                title: 'New Zealand'
+              }
+            ]
+          }
```
Comment (sdirix) [inline packages/material/test/renderers/MaterialOneOfEnumCell.test.tsx:83]: Can't we simplify this?

```suggestion
          country: schema
```

**Review by sdirix — CHANGES_REQUESTED** [review]
Looks good! I only have some minor comments.

Please adapt your commit message. For example you can change it to:

```
Add React Material oneOf enum cell

Also includes example and test cases.
```

Please also add an example in the `examples` package. You can just add an `enumInArray.ts` file with the following content:

```ts
import { registerExamples } from './register';

export const schema = {
  type: 'array',
  items: {
    type: 'object',
    properties: {
      plainEnum: {
        type: 'string',
        enum: ['foo', 'bar']
      },
      oneOfEnum: {
        type: 'string',
        oneOf: [
          { const: 'foo', title: 'Foo' },
          { const: 'bar', title: 'Bar' },
          { const: 'foobar', title: 'FooBar' }
        ]
      },
    }
  }
};

export const uischema = {
  type: 'VerticalLayout',
  elements: [
    {
      type: 'Control',
      scope: '#'
    }
  ]
};

export const data: any[] = [];

registerExamples([
  {
    name: 'enumInArray',
    label: 'Array containing enums',
    data,
    schema,
    uischema
  }
]);
```
Don't forget to also add it to the `index.ts` ;).

**Comment by ryan-bennett** [comment]
Hi Stefan, I think I have processed all of the requested changes. Thanks for walking me though the process.

**Review by sdirix — APPROVED** [review]
LGTM

**Comment by sdirix** [comment]
@ryan-bennett 
JSON Forms `3.0.0-alpha.0` is now released containing the oneOf enum cell renderer. You can consume it by referring to the concrete version or simply use the `next` tag, for example `npm install @jsonforms/core@next @jsonforms/react@next @jsonforms/material-renderers@next`.

Thank you again for the contribution :)

**Pull request merged**